test/fluttertagger_test.dart

```diff
@@ -0,0 +1,272 @@
+
+      await tester.pumpWidget(testWidget);
+
+      final textField = find.byType(TextField);
+      await tester.tap(textField);
+      await tester.pump();
+
+      // Incrementally enter each character to simulate typing
+      await tester.enterText(textField, '@');
+      await tester.pump();
+      await tester.enterText(textField, '@t');
+      await tester.pump();
+      await tester.enterText(textField, '@te');
+      await tester.pump();
+      await tester.enterText(textField, '@tes');
+      await tester.pump();
+      await tester.enterText(textField, '@test');
+      await tester.pumpAndSettle();
+
+      expect(find.text('Overlay'), findsOneWidget);
+    });
+
+    testWidgets('formats text with specific pattern and parser',
+        (tester) async {
+      await tester.pumpWidget(testWidget);
+
+      controller.text =
+          "Hey @11a27531b866ce0016f9e582#brad#. It's time to #11a27531b866ce0016f9e582#Flutter#!";
+      controller.formatTags();
+
+      await tester.pump();
+
+      expect(controller.formattedText,
+          "Hey @11a27531b866ce0016f9e582#brad#. It's time to #11a27531b866ce0016f9e582#Flutter#!");
+      expect(controller.text, "Hey @brad. It's time to #Flutter!");
+    });
+
+    testWidgets('adds tags correctly', (tester) async {
+      await tester.pumpWidget(testWidget);
+
+      // Simulate typing the trigger character and the tag text
+      final textField = find.byType(TextField);
+      await tester.enterText(textField, '@');
+      await tester.pump();
+      await tester.enterText(textField, '@testUser');
+      await tester.pump();
+
+      // Simulate the user selecting a tag from the overlay
+      controller.addTag(id: '123', name: 'testUser');
+      await tester.pump();
+
+      // Verify the formatted text
+      expect(controller.formattedText, '@123#testUser# ');
+    });
+
+    testWidgets('removes tags correctly', (tester) async {
+      await tester.pumpWidget(testWidget);
+
+      // Simulate typing the trigger character and the tag text
+      final textField = find.byType(TextField);
+      await tester.enterText(textField, '@');
+      await tester.pump();
+      await tester.enterText(textField, '@testUser');
+      await tester.pump();
+
+      // Simulate the user selecting a tag from the overlay
+      controller.addTag(id: '123', name: 'testUser');
+      await tester.pump();
+
+      // Simulate the user deleting the tag text character
+      // From actual interactions, 3 backspaces are required to clear
+      await tester.tap(textField);
+      await tester.sendKeyEvent(LogicalKeyboardKey.backspace);
+      await tester.pump();
+      await tester.sendKeyEvent(LogicalKeyboardKey.backspace);
+      await tester.pump();
+      await tester.sendKeyEvent(LogicalKeyboardKey.backspace);
+      await tester.pump();
+
+      // Verify the formatted text is now empty
+      expect(controller.formattedText, '');
+    });
+
+    testWidgets('clears text correctly', (tester) async {
```

Comment:
It would be great to also verify that the controller's `formattedText` is cleared in this test.

Reply:
Updated the test